### PR TITLE
[INLONG-2753][CI] Add support for pushing docker images when building them

### PR DESCRIPTION
**Pull request by shink**
fix: #2753

### Motivation

Add support for pushing the latest docker images when building them.

### Modifications

- `.github/workflows/ci_build_docker.yml`

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### CODE REVIEW

File: .github/workflows/ci_build_docker.yml [hunk 1]
```diff
@@ -37,6 +37,11 @@ jobs:
           java-version: 8
           distribution: adopt
 
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
```
[the remark this answers is not in the document]
Comment (dockerzhang) [reply .github/workflows/ci_build_docker.yml:43]: I have tested `2.7.5`, you can test the `docker/get-project-version.py` using `3.7`.

File: .github/workflows/ci_build_docker.yml [hunk 2]
```diff
@@ -50,3 +55,11 @@ jobs:
...
         env:
           CI: false
+
+      - name: Push docker image to Docker Hub
+        if: ${{ success() && github.event_name == 'push' && github.repository == 'apache/incubator-inlong' }}
+        run: sh docker/publish.sh
+        env:
+          DOCKER_USER: ${{ secrets.DOCKER_USER }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+          DOCKER_REGISTRY: 'hub.docker.com'
```
Comment (dockerzhang) [inline .github/workflows/ci_build_docker.yml:65]: Does the `DOCKER_REGISTRY` is needless, the default registry is `hub.docker.com`.

File: .github/workflows/ci_build_docker.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (shink) [reply .github/workflows/ci_build_docker.yml:43]: I have tested with `3.7.11`, and it worked.

File: .github/workflows/ci_build_docker.yml (hunk 2 above)
Comment (shink) [reply .github/workflows/ci_build_docker.yml:65]: OK, I will remove it.

**Comment by shink** [comment]
@dockerzhang Images are only published when changes are committed to `apache/incubator-inlong`, not if developer commit changes to their own repository or create a PR.

**Comment by shink** [comment]
@dockerzhang OK.

**Comment by dockerzhang** [comment]
image

the `push image` step is skipped?

**Comment by shink** [comment]
> @dockerzhang Images are only published when changes are committed to `apache/incubator-inlong`, not if the developer commits changes to their own repository or create a PR.

@dockerzhang As I said, when the developer commit changes to their own repository or create a PR, the images should not be published. The images will be published when the changes are committed to the only one repository - `apache/incubator-inlong`.

**Review by dockerzhang — APPROVED** [review]

**Review by ciscozhou — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Comment by shink** [comment]
An error occurred while publishing docker images. I have opened another issue #2772 and will fix it.

**Pull request merged**